pkg/awsutils/awsutils.go

```diff
@@ -838,33 +852,109 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 			SecondaryPrivateIpAddressCount: aws.Int64(int64(needIPs)),
 		}
 	}
-	if useCustomCfg {
-		log.Info("Using a custom network config for the new ENI")
-		if len(sg) != 0 {
-			input.Groups = sg
-		} else {
-			log.Warnf("No custom networking security group found, will use the node's primary ENI's SG: %v", aws.StringValueSlice(input.Groups))
+
+	var err error
+	if cache.useCustomNetworking {
+		input = createEniUsingCustomCfg(sg, customSubnet, input)
+		log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+
+		start := time.Now()
+		result, err = cache.ec2SVC.CreateNetworkInterfaceWithContext(context.Background(), input)
+		prometheusmetrics.Ec2ApiReq.WithLabelValues("CreateNetworkInterface").Inc()
+		prometheusmetrics.AwsAPILatency.WithLabelValues("CreateNetworkInterface", fmt.Sprint(err != nil), awsReqStatus(err)).Observe(msSince(start))
+		if err == nil {
+			log.Infof("Created a new ENI: %s", aws.StringValue(result.NetworkInterface.NetworkInterfaceId))
+			return *result.NetworkInterface.NetworkInterfaceId, nil
 		}
-		input.SubnetId = aws.String(subnet)
+		checkAPIErrorAndBroadcastEvent(err, "ec2:CreateNetworkInterface")
+		awsAPIErrInc("CreateNetworkInterface", err)
+		prometheusmetrics.Ec2ApiErr.WithLabelValues("CreateNetworkInterface").Inc()
+		log.Errorf("Failed to CreateNetworkInterface %v for subnet %s", err, customSubnet)
 	} else {
-		log.Info("Using same config as the primary interface for the new ENI")
+		subnetResult, vpcErr := cache.getVpcSubnets()
+		if vpcErr != nil {
+			return "", errors.Wrap(vpcErr, "failed to get subnets in vpc")
+		}
+
+		for _, subnet := range subnetResult {
+			if *subnet.SubnetId != cache.subnetID {
+				if !validTag(subnet) {
```

Comment:
What happens if we don't have the validTag at all on all subnets? Maybe someone removed it..

Reply:
Assuming the default subnet is out of IPs, it won't create an ENI and will try again later. Same as previous logic